src/server/utils/cluster-manager/cluster-manager.ts

```diff
@@ -370,9 +386,25 @@ export class ClusterManager {
 
     logger.log(`Introspecting all sources in cluster '${cluster.name}'`);
 
-    return Promise.all(this.managedExternals.map(managedExternal => {
-      return this.introspectManagedExternal(managedExternal);
-    })).then(noop);
+    return (External.getConstructorFor(cluster.type) as any).getSourceList(this.requester)
+        .then(
+            (sources: string[]) => {
+              let introspectionTasks: Array<Promise<void>> = [];
+              sources.forEach(source => {
+                const existingExternalsForSource = this.managedExternals.filter(managedExternal => getSourceFromExternal(managedExternal.external) === source);
```

Comment:
When source could have more than one external?

Reply:
I wondered about it as well but I couldn't find the answer.
The method which it calls is `getSourceList` which comes from plywood lib 
(https://github.com/implydata/plywood/blob/master/src/external/druidExternal.ts)
Keep in mind that this method was used in the code earlier and not something new.